metabox/metabox/core/scenario.py

```diff
@@ -63,10 +70,15 @@ def __init__(self, mode, *releases, remote_revision="current", service_revision=
         self.service_revision = service_revision
         self._checks = []
         self._ret_code = None
-        self._stdout = ''
-        self._stderr = ''
+        self._stdout = ""
+        self._stderr = ""
         self._pts = None
 
+    def get_output_streams(self):
+        if self._pts:
+            return self._pts.stdout_data_full.decode("utf-8")
+        return self._stdout + self._stderr
```

Comment:
This is almost never a good idea. Those two streams have their own timeline, and concatenating them like that can be very misleading. Either:
1. when grabbing the streams, create a mixed one (while preserving the discrete ones, or not :) )
2. yield them separately.

Reply:
I was being lazy here because I did not want to overhaul the `run_or_raise` api. You are right, I have changed this. Also: we should really look into that function and at least change the name